src/LuaTransformer.ts

```diff
@@ -2628,9 +2628,20 @@ export class LuaTransformer {
     }
 
     public transformForInStatement(statement: ts.ForInStatement): StatementVisitResult {
-        // Get variable identifier
-        const variable = (statement.initializer as ts.VariableDeclarationList).declarations[0];
-        const identifier = variable.name as ts.Identifier;
+        // Transform iteration variable
+        let iterationVariable: tstl.Identifier;
+        if (
+            ts.isVariableDeclarationList(statement.initializer) &&
+            !ts.isArrayBindingPattern(statement.initializer.declarations[0].name) &&
+            !ts.isObjectBindingPattern(statement.initializer.declarations[0].name)
+        ) {
+            iterationVariable = this.transformIdentifier(statement.initializer.declarations[0].name);
+        } else if (ts.isIdentifier(statement.initializer)) {
+            iterationVariable = this.transformIdentifier(statement.initializer);
```

Comment:
That way it would create a new variable instead of reassigning existing one. It should do it similarly to for...of:
```lua
for ____key in pairs(expression) do
    initializer = ____key
end
```

Reply:
I'm not sure what you mean, line 260 will just directly translate the variable name, `for <x> in y` -> `for <x> in pairs(y)`?